### PR TITLE
feat: move contracts

### PR DESCRIPTION
**Pull request by heueristik**
This PR moves the contracts and adapts the unit and integration tests.

Task ID: [OS-473](https://aragonassociation.atlassian.net/browse/OS-473)

Tests will be improved in an upcoming task [OS-1093](https://aragonassociation.atlassian.net/browse/OS-1093).

[OS-473]: https://aragonassociation.atlassian.net/browse/OS-473?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[OS-1093]: https://aragonassociation.atlassian.net/browse/OS-1093?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/contracts/test/test-utils/typechain-versions.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+/// Typechain will sometimes by default link to the wrong version of the contract, when we have name collisions
+/// The version specified in src is the factory and contract without the version number.
+/// Import as needed in the test files, and use the correct version of the contract.
+
+export {Multisig__factory as Multisig_V1_0_0__factory} from '../../typechain/factories/@aragon/osx-v1.0.0/plugins/governance/multisig/Multisig__factory';
+export {Multisig__factory as Multisig_V1_3_0__factory} from '../../typechain/factories/@aragon/osx-v1.3.0/plugins/governance/multisig/Multisig__factory';
+export {Multisig__factory} from '../../typechain/factories/src/Multisig__factory';
+export {Multisig} from '../../typechain/src/Multisig';
+export {IMultisig} from '../../typechain/src/IMultisig';
```
Comment (heueristik) [inline packages/contracts/test/test-utils/typechain-versions.ts:9]: We have to understand why this is required

File: packages/contracts/src/Multisig.sol [hunk 2]
```diff
@@ -6,6 +6,7 @@ import {SafeCastUpgradeable} from "@openzeppelin/contracts-upgradeable/utils/mat
 
 import {IMembership} from "@aragon/osx-commons-contracts/src/plugin/extensions/membership/IMembership.sol";
 import {Addresslist} from "@aragon/osx-commons-contracts/src/plugin/extensions/governance/Addresslist.sol";
+// solhint-disable-next-line max-line-length
```
Comment (jordaniza) [inline packages/contracts/src/Multisig.sol]: I think the enable/disable around the whole imports section is personally much cleaner than on a per line basis BUT that's a minor and personal preference. You can ignore if you wish

File: packages/contracts/test/20_integration-testing/test-helpers.ts [hunk 3]
```diff
@@ -213,3 +216,31 @@ async function checkPermissions(
     throw `The used signer does not have the permission with ID '${applyPermissionId}' granted and thus cannot apply the setup`;
   }
 }
+
+// TODO Move into OSX commons ?
```
Comment (jordaniza) [inline packages/contracts/test/20_integration-testing/test-helpers.ts]: Agreed, to be added to our cleanup task

File: packages/contracts/test/test-utils/typechain-versions.ts (hunk 1 above)
Comment (jordaniza) [reply packages/contracts/test/test-utils/typechain-versions.ts:9]: If you try and import from the `index.ts` file at the root of the autogenerated `typechain` folder, then you will see that, when we import multiple versions of the same named contract, typechain doesn't export the correct one (at least in my case with the tokenVoting). Specifying a particular version in this way ensures you use the correct version of the contracts at all times.

**Review by jordaniza — DISMISSED** [review]

File: packages/contracts/src/Multisig.sol (hunk 2 above)
Comment (heueristik) [reply packages/contracts/src/Multisig.sol]: Yes, I thought the same. Changed in 003091042abddd400ef3225e42ba70c21e259c78

File: packages/contracts/test/20_integration-testing/test-helpers.ts (hunk 3 above)
Comment (heueristik) [reply packages/contracts/test/20_integration-testing/test-helpers.ts]: Added the task ID in e7126a0d96ef8c16344d39d2a6e30eecfa062756

**Review by jordaniza — APPROVED** [review]

**Pull request merged**